### PR TITLE
Refactoring tests

### PR DESCRIPTION
**Pull request by carusogabriel**
Continuing to refactory tests, like #1676, I used:
- `assertCount` instead of `count` function;
- `assertInstanceOf` instead of `instanceof` operator;
- `assertArrayHasKey` instead of `array_key_exists` function;
- `assertInternalType` instead of `is_*` functions;
- `assertFalse` instead of strict comparisons with `false` keyword;
- `assertGreaterThan` for mathematical comparisons;
- `assertSame` and `assertNotSame` instead of strict comparisons;
- `assertContains` instead of `in_array` function;
- `assertEmpty` and `assertNotEmpty` instead of `empty` function.

### CODE REVIEW

File: tests/Doctrine/ODM/MongoDB/Tests/Aggregation/Stage/GraphLookupTest.php [hunk 1]
```diff
@@ -90,7 +90,7 @@ public function testGraphLookupWithEmployees(\Closure $addGraphLookupStage, arra
                 continue;
             }
 
-            $this->assertGreaterThan(0, count($reportingHierarchy->reportingHierarchy));
+            $this->assertGreaterThan(0, $reportingHierarchy->reportingHierarchy);
```
Comment (alcaeus) [inline tests/Doctrine/ODM/MongoDB/Tests/Aggregation/Stage/GraphLookupTest.php]: I believe this change is incorrect: greaterThan compares expected and actual directly and doesn't account for countables.

Comment (carusogabriel) [reply tests/Doctrine/ODM/MongoDB/Tests/Aggregation/Stage/GraphLookupTest.php]: You are correct, assertGreaterThan just [check the mathematical condition](https://github.com/sebastianbergmann/phpunit/blob/master/src/Framework/Constraint/GreaterThan.php#L43), it doesn't count.

**Comment by carusogabriel** [comment]
I've made the requested changes, and, use `assertStringEqualsFile` when comparing a `string` and a file.

**Comment by carusogabriel** [comment]
I've refactored more tests using `assertArrayHasKey` and `assertArrayNotHasKey` instead of checking the key with isset function.

**Comment by alcaeus** [comment]
Thanks @carusogabriel!

**Pull request merged**